### PR TITLE
KCI-135 > integrate SVG icons (free and pro)

### PR DESCRIPTION
**Pull request by jvhellemondt**
Free svg works and pro already worked :100: 

image

image
image

### CODE REVIEW

File: generator/templates/Default/src/plugins/vuetify/FontawesomePro.js [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+  viewList: ['far', 'list'],
+  viewCalendar: ['far', 'calendar-alt'],
+  viewChart: ['far', 'chart-bar'],
+
+  genderUnknown: ['far', 'genderless'],
+  genderMale: ['far', 'mars'],
+  genderFemale: ['far', 'venus'],
+
+  lock: ['fas', 'lock-alt'],
+  unlock: ['fas', 'unlock-alt'],
```
Comment (ricardovanlaarhoven) [inline generator/templates/Default/src/plugins/vuetify/FontawesomePro.js]: This is project specific. Not for the scaffold.

File: generator/templates/Default/src/plugins/vuetify/FontawesomePro.js [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+  hexagon: ['fas', 'hexagon'],
+
+  car: ['far', 'car'],
+  train: ['far', 'train'],
+
+  phone: ['far', 'phone'],
+  email: ['far', 'envelope'],
+  fax: ['far', 'fax'],
+  website: ['far', 'globe-europe'],
+  address: ['far', 'map-marker-alt'],
```
Comment (ricardovanlaarhoven) [inline generator/templates/Default/src/plugins/vuetify/FontawesomePro.js]: This is project specific. Not for the scaffold.

File: generator/templates/Default/src/plugins/vuetify/FontawesomePro.js [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+
+  profile: ['far', 'user'],
+  settings: ['far', 'cog'],
+  home: ['far', 'home'],
+  notifications: ['far', 'bell'],
+  search: ['far', 'search'],
+  birthday: ['far', 'birthday-cake'],
+  invited: ['far', 'hourglass-half'],
+  unknown: ['far', 'question-circle'],
+  fabDots: ['fas', 'ellipsis-v'],
```
Comment (ricardovanlaarhoven) [inline generator/templates/Default/src/plugins/vuetify/FontawesomePro.js]: This is project specific. Not for the scaffold.

File: generator/templates/Default/src/plugins/vuetify/FontawesomePro.js [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+
+  vacancy: ['far', 'handshake'],
+  shift: ['far', 'business-time'],
+  candidate: ['far', 'users'],
+  employer: ['far', 'briefcase'],
+  report: ['far', 'chart-bar'],
+  role: ['far', 'user-unlock'],
+  location: ['far', 'building'],
+  department: ['far', 'building'],
+  position: ['far', 'briefcase'],
```
Comment (ricardovanlaarhoven) [inline generator/templates/Default/src/plugins/vuetify/FontawesomePro.js]: This is project specific. Not for the scaffold.

File: generator/templates/Default/src/plugins/vuetify/FontawesomePro.js [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+
+  upload: ['far', 'upload'],
+  fileUnknown: ['far', 'file'],
+  fileVideo: ['far', 'file-video'],
+  fileWord: ['far', 'file-word'],
+  fileExcel: ['far', 'file-excel'],
+  filePowerpoint: ['far', 'file-powerpoint'],
+  filePdf: ['far', 'file-pdf'],
+  folderOpen: ['far', 'folder-open'],
+  folderClosed: ['far', 'folder'],
```
Comment (ricardovanlaarhoven) [inline generator/templates/Default/src/plugins/vuetify/FontawesomePro.js]: Is this global? I'm not sure but i don't think so.

File: generator/templates/Default/src/plugins/vuetify/theme.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+const light = {
+  primary: '#2A73C5',
+};
+
+const dark = {
+  primary: '#2A73C5',
+};
```
Comment (ricardovanlaarhoven) [inline generator/templates/Default/src/plugins/vuetify/theme.js]: Lucky i saw this.
Those are not the same color as they where.
The previous colors are the Kings Code colors and the default for the scaffold.

**Review by ricardovanlaarhoven — CHANGES_REQUESTED** [review]

File: generator/templates/Default/src/plugins/vuetify/FontawesomePro.js (hunk 5 above)
Comment (jvhellemondt) [reply generator/templates/Default/src/plugins/vuetify/FontawesomePro.js]: Not sure what you mean: these are used in the scaffold. And yes, it is available as `<v-icon>$****</v-icon>`

Comment (ricardovanlaarhoven) [reply generator/templates/Default/src/plugins/vuetify/FontawesomePro.js]: The same as the other comments, except for this one i'm not sure.
So are these icons supposed to be global/in the scaffold?
I'm not sure but i don't think so.

**Review by martn001 — APPROVED** [review]

**Review by ricardovanlaarhoven — APPROVED** [review]

File: generator/templates/Default/src/plugins/vuetify/FontawesomePro.js (hunk 5 above)
Comment (jvhellemondt) [reply generator/templates/Default/src/plugins/vuetify/FontawesomePro.js]:
```
  fileUnknown: ['far', 'file'],
  fileVideo: ['far', 'file-video'],
  fileWord: ['far', 'file-word'],
  fileExcel: ['far', 'file-excel'],
  filePowerpoint: ['far', 'file-powerpoint'],
  filePdf: ['far', 'file-pdf'],
```

@ricardovanlaarhoven , not sure if this is still a point of discussion.
Excel and powerpoint could be dropped, these are not used. The others are actually used in KFileField. Removing those will result in a broken component.

**Pull request merged**